app/routes/u.$identifier/results/index.tsx

```diff
@@ -27,7 +27,8 @@ export default function UserResultsPage() {
   const isOwnResultsPage = user?.id === userPageData.id;
 
   const showHighlightsSection =
-    hasHighlights || (isOwnResultsPage && hasResults);
+    (hasHighlights || (isOwnResultsPage && hasResults)) &&
+    userPageData.results.length !== 1; // Hide Highlights section if the user only has 1 result
```

Comment:
Thanks for looking at this!

I believe this logic might lead to issues with existing users who only have one result, which they've highlighted.
If these users exist, I believe the result would effectively be hidden until they get an additional result.

How about changing it, so that the highlight section is still always visible when `hasHighlights` is true? 🙂

Reply:
Oh I see, I thought you wanted the entire Highlights section gone & not just the `Choose Highlights` button. I'll make the adjustment now that you've clarified it for me :)

After adjusting it, here are the scenarios:

- 0 Highlights, 1 Result: the `Choose Highlights` button is displayed:
![image](https://user-images.githubusercontent.com/15804376/198854885-e44d8142-0bfe-42d8-8fd5-0de79e2b4309.png)

- 1 Highlight, 1 Result (though the Result is hidden once it's been highlighted, it is still in the results array): the `Choose Highlights` button is no longer there:

![image](https://user-images.githubusercontent.com/15804376/198854903-b5d2dc37-ce66-47de-b9ab-b89332359a99.png)

Any other scenario that doesn't match the other 2 options: the `Choose Highlights` button should be displayed (although I can't test this yet without figuring out how to add a new result to the database)

Reply:
Okay, I managed to tinker with the DB & insert some data in the appropriate tables to get this test case:

- 1 Highlight, 2 total results (aka, 1 non-highlight): `Choose Highlights` button still shows:

![image](https://user-images.githubusercontent.com/15804376/198855569-524aba24-4f15-431c-bd3d-129839203681.png)

Reply:
Oh, sorry, I think there was a misunderstanding.
If I understood @Sendouc correctly here, this issue is about: not showing the highlight section (including the choose highlights button) to a user if they have only one event result.

We basically want to disencourage people from marking a result as a "highlight", if it's their only one.

So your original solution was basically doing that.
My suggestion was to then still allow everyone to see your highlight (and edit it if it's your own), if you in the past did exactly the thing we're trying to prevent here.

So it's not just about the button, In my head I was thinking of a change from 
```typescript
const showHighlightSection = (hasHighlights || (isOwnResultsPage && hasResults)) && userPageData.results.length !== 1;
```
to:
```typescript
const showHighlightSection = hasHighlights || (isOwnResultsPage && hasResults && userPageData.results.length !== 1);
```

    
Note that we could additionally enforce all of these things on the backend side of things and also edit the DB to migrate everyone who has this scenario. But personally I don't see a need for that. If I got it correctly it's mostly about disencouragement of misusing this feature. @Sendouc?

Reply:
Okay I see, my bad - yeah this one was harder for me to understand the intention of without additional info (and also I didn't initially see that the non-highlighted results does not equal the total results until much later on). Luckily, the fix is relatively simple & is just a one-liner like you mentioned :)

Reply:
This is now what the page looks like if the user only has 1 total result:

![image](https://user-images.githubusercontent.com/15804376/198881927-f7a9753f-ee3b-4130-b20b-ca505d2b1360.png)

If for some reason they had already highlighted their result, it will still render properly due to `hasHighlights` being true